include/sparrow_v01/layout/array_base.hpp

```diff
@@ -66,7 +64,7 @@ namespace sparrow
      * traits class that must be specialized by array
      * classes inheriting from array_crtp_base.
      */
-    template <class D>
+    template <typename D>
```

Comment:
this makes reviewing harder than it should be.
If you want to make these cosmetic changes make them in a separate PR